web/src/user_profile.ts

```diff
@@ -482,6 +482,33 @@ export function get_custom_profile_field_data(
             profile_field.value = field_choice_dict[field_value.value]!.text;
             break;
         }
+        case field_types.SELECT_MULTIPLE.id: {
+            try {
+                const field_data_schema = z.record(z.string(), z.object({text: z.string()}));
+                const field_data = field_data_schema.parse(JSON.parse(field.field_data));
+                const selected_ids = z.array(z.string()).parse(JSON.parse(field_value.value));
+                const readable_values: string[] = [];
+                for (const id of selected_ids) {
+                    const option = field_data[id];
+                    if (option) {
+                        readable_values.push(option.text);
+                    }
+                }
+                const final_text = readable_values.join(", ");
```

Comment:
I think you want to use one of the helpers from `util.ts` that use `Intl.ListFormat` to do this in an internationalized fashion. (Japanese doesn't use `, ` as the separator, for example).